test/rom_num_router_test.exs

```diff
@@ -0,0 +1,47 @@
+defmodule RomNumRouterTest do
+  use ExUnit.Case
+  use Plug.Test
+
+  @opts RomNumRouter.init([])
+
+  test "goes to index page" do
+    conn = conn(:get, "/", TemplateDisplay.template_index)
+    conn = RomNumRouter.call(conn, @opts)
+
+    assert conn.state == :sent
+    assert conn.status == 200
+  end
+
+  test "go to Arabic numbers page" do
+    conn = conn(:post, "/arabic_numbers", "")
+    conn = RomNumRouter.call(conn, @opts)
+
+    assert conn.state == :sent
+    assert conn.status == 200
+  end
+
+  test "if page not found return 404" do
+    conn = conn(:get, "/weird_request", "Sorry, page not found")
+    conn = RomNumRouter.call(conn, @opts)
+
+    assert conn.state == :sent
+    assert conn.status == 404
+
+  end
+
+  test "index page displays eex html template as response body" do
+    conn = conn(:get, "/", TemplateDisplay.template_index)
+    conn = RomNumRouter.call(conn, @opts)
+
+    assert conn.resp_body =~ "Arabic Number"
+  end
+
+  test "input from html form is given in response body" do
+    conn = conn(:post, "/arabic_numbers", "arabic_number=6")
+    conn = RomNumRouter.call(conn, @opts)
+    {:ok, body, _} = read_body(conn)
```

Comment:
Why does this test call `read_body/3` while the previous test gets the body from `conn.resp_body` ? Are they different?